src/scss/base.scss

```diff
@@ -189,6 +194,78 @@
 		@include oFormsGroupInlineRadioCheckbox($class);
 	}
 }
+/// @access public
+/// @param {string} $class - The block level class name (BEM naming convention).
+/// @output Styles for form status such as "saving" and "saved".
+@mixin oFormsStatusFeature($class: 'o-forms') {
+	// Icons align to a 40x40 pixel grid with 10px padding on each side.
+	// In this case however we want to remove that padding to align the icon flush with the input.
+	// https://github.com/Financial-Times/fticons/blob/master/contributing.md#design
+	$status-icon-size: $_o-forms-field-small-height;
+	$status-icon-padding: ($status-icon-size/40) * 10;
+	@include oBrandConfigureFor($component: 'o-forms') {
+		// Base status styles.
+		.#{$class}__status {
+			@include oTypographySize($_o-forms-typography-scale-small);
+			opacity: 0;
+			visibility: hidden;
+			display: flex;
+			white-space: nowrap;
+			align-items: center;
+			min-width: 5em;
+			&:after {
+				content: '';
+			}
+			&:before {
+				content: '';
+				margin: #{-$status-icon-padding} 0 #{-$status-icon-padding} #{-$status-icon-padding};
+			}
+		}
+
+		.#{$class}__status[aria-hidden="false"] {
+			display: flex;
+			visibility: visible;
+			opacity: 1;
+		}
+
+		// Saving status.
+		.#{$class}--saving .#{$class}__status {
+			&:after {
+				content: 'Saving';
+			}
+			&:before {
+				@include oLoading();
+				@include oLoadingSize('small');
+				@include oLoadingColor('dark');
+				// Reszise loader to match icon.
+				margin: 0 $status-icon-padding 0 0;
+				width: 10px;
+				height: 10px;
+				border-width: 2px;
+			}
+		}
+
+		// Saved status.
+		.#{$class}--saved .#{$class}__status {
+			&:after {
+				content: 'Saved';
```

Comment:
I'm confused why it's setting content with a pseudo-element, can't the label sit inside the status element?

Reply:
It could, but this way helps keep language consistent across apps and makes it a little easier to implement as you don't have to worry about updating the copy between states.